### DISCUSSION TITLE
请问如何改善V2Ray的延迟？

### DISCUSSION THREAD

**Discussion by imotosaiko** [v2fly/v2ray-core#2338] [Q&A]
使用V2Ray自建代理服务器，但是发现有一点点延迟的问题
下图是使用ping命令测得的延迟
![图片](https://user-images.githubusercontent.com/71994020/219865703-98b7edf4-c432-4b45-b765-059ac3fa5841.png)
下图是在服务器上安装Nginx后，在浏览器用Firefox中测得的访问nginx服务器的延迟
![图片](https://user-images.githubusercontent.com/71994020/219865720-89307c80-01f7-4aa7-976e-98946d5cfd71.png)
然后这个是我使用v2raya测得的延迟
![图片](https://user-images.githubusercontent.com/71994020/219865405-ab88833e-cccb-474b-adcd-337b2066deba.png)
作为参考，我正在使用的梯子的游戏线路的延迟在150ms左右（使用Clash For Windows测得，因为哪个梯子只能在这里用游戏线路），该游戏线路和我的服务器都是位于日本的
起初我以为是不是使用了vmess协议的问题，因为我注意到哪个梯子的游戏线路都是shadowsocks的协议，但是今天我尝试了shadowsocks后发现延迟并没有明显的变动
![图片](https://user-images.githubusercontent.com/71994020/219865533-9d15bd34-c5e4-400a-bf17-fa900c7a9f92.png)
所以想请问一下是什么会造成延迟的问题呢？比如使用udp代替tcp？更好的配置文件？或者这是不可避免的，我只能换一台网络延迟更低的服务器才能解决呢？
望大神看到这里可以不吝赐教！万分感谢

**Comment by mydogshitgold** [comment]
Did you tried to enable `mux`? 

Example:
```jsonc
{
  "protocol": "vmess",
  // ...
  "mux": {
    "enabled": true
  }
}

```

**Reply by imotosaiko** [reply]
Thanks!I will try it.

**Comment by IRN-Kawakaze** [comment]
HTTP延迟和实际游戏延迟不一样。对于HTTP流量，两端的routing以及服务端的freedom出站都有可能需要进行dns解析，若不使用v2ray内置的DNS模块，这几处会各增加1*dns解析时间的延迟。另外HTTP握手本身也有多个RTT，比ping延迟（1-RTT）高很多很正常。部分使用UDP的游戏（比如FPS游戏）不存在上述问题，即便看起来HTTP延迟非常高，实际游戏内延迟≈ping延迟。

FPS游戏建议直接看游戏内延迟，忽略HTTP延迟。

如果想降低dns解析产生的延迟，一个缓解的办法是出站的domainStrategy使用UseIP，并且不禁用v2ray的dns模块的缓存，这样routing时使用的解析结果可以复用到freedom出站，甚至解析结果可以复用到后续的连接。另外换一个延迟足够低的dns也会很有效果。

如果延迟大头不是dns解析（通常是这种情况），并且还想降低HTTP延迟，可以尝试多路复用，比如mux.cool、gRPC或者QUIC。（对于FPS游戏，它们反而可能会产生副作用）

**Reply by imotosaiko** [reply]
原来如此，明白了！

**Comment by ghost** [comment]
这延迟正常吧。假设 dns 请求在服务端完成，一般服务器的 dns 解析延迟在 15 ms 以下。tcp 的 RTT 加上 vmess 请求，大约消耗 3个 RTT。取平均 RTT 68 ms 算，68 \* 3 = 204。服务器到测延迟的网站的延迟不超过 20 ms，加上网络抖动，这个延迟完全正常，甚至由于能达到理论值，某种意义上来说算是优秀。

**Comment by 1265578519** [comment]
你需要使用这款工具来测试游戏延迟，它使用了真实的udp传输数据来模拟游戏流量。
这是香港延迟，非常适合游戏，一直用来代替UU加速器玩cod16，游戏内延迟40-44ms左右。
![image](https://user-images.githubusercontent.com/6442439/224564245-3908aba6-d777-491e-a053-b2f972e311a7.png)

![image](https://user-images.githubusercontent.com/6442439/224564256-9ee94c95-0e97-4c83-8fc8-cbdf08b5cf07.png)

[sddt_win64_2.zip](https://github.com/v2fly/v2ray-core/files/10951643/sddt_win64_2.zip)

**Comment by shentong0722** [comment]
我也遇到了类似的问题，（怀疑是延迟问题但是不确定），自建的VMESS节点，VPS配置2核2G绝对够用，我的网络环境是电信1000Mbps宽带，VPS是万兆，用speedtest测速基本都能跑满800+速度说明没有问题，但是打开网页白屏时间很长，真的特别长，打开之后速度也没问题，比如看youtube 8K没有任何卡顿

**Reply by Razorro** [reply]
那应该是DNS的问题吧

**Reply by shentong0722** [reply]
考虑过DNS的问题，不过有个点我想不明白，理论上用代理访问境外的网站应该是用境外的DNS更快，比如8.8.8.8,1.1.1.1等，但是实测都不如阿里223.5.5.5快，这有点奇怪

**Reply by toney871030** [reply]
是不是有点来到我的地盘还得听我的这种感觉，此树是我栽

**Comment by lfbxindekaishi** [comment]
通过V2ray连接服务器后，真实延迟基本都能200左右，但是网速时快时慢。 流畅的时候Youtebe视频1080秒开，卡的时候网页都打不开。服务器的设置一直都没变过，v2ray的设置也都是一样没变过，为什么网速会出现这样的情况呢？
